src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let result = { ...state };
+  const resArray = [];
+
+  for (const act of actions) {
+    if (act.type === 'addProperties') {
+      Object.assign(result, act.extraData);
+      resArray.push({ ...result });
+      continue;
```

Comment:
remove continue, you've already have if block, if the condition hasn;t met the code won't other if-blocks